File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/pojo/FeatSpecialRequirement.java
package ai.ragu.charactermanager.pojo;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class FeatSpecialRequirement {

    private long id;
    private String featId;
    private String featCategory;
    private String requiredSpecialId;
    private String requiredSpecialCategory;

}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/pojo/FeatSpecialRequirement.kt
package ai.ragu.charactermanager.pojo

import AbstractJpaPersistable
import javax.persistence.*

@Entity
@Table(name = "feat_special_requirement")
class FeatSpecialRequirement : AbstractJpaPersistable<Long>() {

    @Column
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "feat_id")
    lateinit var feat: Feat

    @Column
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "special_id")
    lateinit var special: Special
}